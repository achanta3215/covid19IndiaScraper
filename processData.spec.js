const { transformStatesData } = require('./processData');

const mockData = [
  {
    state: 'Andaman and Nicobar Islands',
    active: '46',
    discharged: '130',
    deceased: '0',
    total: '176',
  },
  {
    state: 'Andhra Pradesh',
    active: '16621',
    discharged: '18378',
    deceased: '452',
    total: '35451',
  },
];

const nextDayMockData = mockData.map((data) => ({
  state: data.state,
  active: String(Number(data.active) + 50),
  discharged: String(Number(data.discharged) + 20),
  deceased: String(Number(data.deceased) + 5),
  total: String(Number(data.total) + 50),
}));

const thirdDayMockDate = nextDayMockData.map((data) => ({
  state: data.state,
  active: String(Number(data.active) + 30),
  discharged: String(Number(data.discharged) + 10),
  deceased: String(Number(data.deceased) + 2),
  total: String(Number(data.total) + 25),
}));

describe('Processes input correctly', () => {
  it('Stores correctly for the first time', () => {
    const date = new Date(1594970387627);
    const newStatsObj = transformStatesData(mockData, date, {});
    expect(newStatsObj).toMatchSnapshot();
  });
  it('Appends statsObj for the next day when previous day has no delta', () => {
    const date = new Date(1594970387627);
    const dateTomorrow = new Date(1595058097304);
    const firstDayStatsObj = transformStatesData(mockData, date, {});
    const appendedStatsObj = transformStatesData(
      nextDayMockData,
      dateTomorrow,
      firstDayStatsObj,
    );
    expect(appendedStatsObj).toMatchSnapshot();
  });
  it('Appends statsObj for the next day when previous day has delta', () => {
    const date = new Date(1594970387627);
    const dateTomorrow = new Date(1595058097304);
    const dateThirdDay = new Date(1595152388171);
    const firstDayStatsObj = transformStatesData(mockData, date, {});
    const secondDayStatsObj = transformStatesData(
      nextDayMockData,
      dateTomorrow,
      firstDayStatsObj,
    );
    const thirdDayStatsObj = transformStatesData(
      thirdDayMockDate,
      dateThirdDay,
      secondDayStatsObj,
    );
    expect(thirdDayStatsObj).toMatchSnapshot();
  });
});
