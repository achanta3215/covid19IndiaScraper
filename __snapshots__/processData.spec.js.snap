// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processes input correctly Appends statsObj for the next day when previous day has delta 1`] = `
Object {
  "07172020": Object {
    "Andaman and Nicobar Islands": Object {
      "active": "46",
      "deceased": "0",
      "discharged": "130",
      "state": "Andaman and Nicobar Islands",
      "total": "176",
    },
    "Andhra Pradesh": Object {
      "active": "16621",
      "deceased": "452",
      "discharged": "18378",
      "state": "Andhra Pradesh",
      "total": "35451",
    },
  },
  "07182020": Object {
    "Andaman and Nicobar Islands": Object {
      "active": "96",
      "deceased": "5",
      "delta": Object {
        "active": "50",
        "deceased": "5",
        "discharged": "20",
        "state": "Andaman and Nicobar Islands",
        "total": "50",
      },
      "discharged": "150",
      "state": "Andaman and Nicobar Islands",
      "total": "226",
    },
    "Andhra Pradesh": Object {
      "active": "16671",
      "deceased": "457",
      "delta": Object {
        "active": "50",
        "deceased": "5",
        "discharged": "20",
        "state": "Andhra Pradesh",
        "total": "50",
      },
      "discharged": "18398",
      "state": "Andhra Pradesh",
      "total": "35501",
    },
  },
  "07192020": Object {
    "Andaman and Nicobar Islands": Object {
      "active": "126",
      "deceased": "7",
      "delta": Object {
        "active": "30",
        "deceased": "2",
        "discharged": "10",
        "state": "Andaman and Nicobar Islands",
        "total": "25",
      },
      "discharged": "160",
      "state": "Andaman and Nicobar Islands",
      "total": "251",
    },
    "Andhra Pradesh": Object {
      "active": "16701",
      "deceased": "459",
      "delta": Object {
        "active": "30",
        "deceased": "2",
        "discharged": "10",
        "state": "Andhra Pradesh",
        "total": "25",
      },
      "discharged": "18408",
      "state": "Andhra Pradesh",
      "total": "35526",
    },
  },
}
`;

exports[`Processes input correctly Appends statsObj for the next day when previous day has no delta 1`] = `
Object {
  "07172020": Object {
    "Andaman and Nicobar Islands": Object {
      "active": "46",
      "deceased": "0",
      "discharged": "130",
      "state": "Andaman and Nicobar Islands",
      "total": "176",
    },
    "Andhra Pradesh": Object {
      "active": "16621",
      "deceased": "452",
      "discharged": "18378",
      "state": "Andhra Pradesh",
      "total": "35451",
    },
  },
  "07182020": Object {
    "Andaman and Nicobar Islands": Object {
      "active": "96",
      "deceased": "5",
      "delta": Object {
        "active": "50",
        "deceased": "5",
        "discharged": "20",
        "state": "Andaman and Nicobar Islands",
        "total": "50",
      },
      "discharged": "150",
      "state": "Andaman and Nicobar Islands",
      "total": "226",
    },
    "Andhra Pradesh": Object {
      "active": "16671",
      "deceased": "457",
      "delta": Object {
        "active": "50",
        "deceased": "5",
        "discharged": "20",
        "state": "Andhra Pradesh",
        "total": "50",
      },
      "discharged": "18398",
      "state": "Andhra Pradesh",
      "total": "35501",
    },
  },
}
`;

exports[`Processes input correctly Stores correctly for the first time 1`] = `
Object {
  "07172020": Object {
    "Andaman and Nicobar Islands": Object {
      "active": "46",
      "deceased": "0",
      "discharged": "130",
      "state": "Andaman and Nicobar Islands",
      "total": "176",
    },
    "Andhra Pradesh": Object {
      "active": "16621",
      "deceased": "452",
      "discharged": "18378",
      "state": "Andhra Pradesh",
      "total": "35451",
    },
  },
}
`;
